### PR TITLE
Use closure for navigation. 

### LINKED ISSUE #486: Only generate routes when required for navigation
See https://mailman.owncloud.org/pipermail/devel/2015-March/001146.html for more infos

### PR DESCRIPTION
**Pull request by VicDeo**
Fixes #486

### CODE REVIEW

**Comment by VicDeo** [comment]
@nickvergessen FYI

File: appinfo/app.php [hunk 1]
```diff
@@ -21,61 +21,72 @@
...
+\OCP\App::registerPersonal('documents', 'personal');
 
-OCP\App::addNavigationEntry(array(
-	'id' => 'documents_index',
-	'order' => 2,
-	'href' => OCP\Util::linkTo('documents/', 'index.php'),
-	'icon' => OCP\Util::imagePath('documents', 'documents.svg'),
-	'name' => OCA\Documents\Config::getL10n()->t('Documents'))
-);
+$navigationEntry = function () use ($c) {
```
Comment (nickvergessen) [inline appinfo/app.php]: `use ($c)` but you dont use it?

File: appinfo/app.php [hunk 2]
```diff
@@ -21,61 +21,72 @@
...
+	return [
+		'id' => 'documents_index',
+		'order' => 2,
+		'href' => \OCP\Util::linkTo('documents/', 'index.php'),
+		'icon' => \OCP\Util::imagePath('documents', 'documents.svg'),
+		'name' => Config::getL10n()->t('Documents')
```
Comment (nickvergessen) [inline appinfo/app.php]: this sounds fishy wrond

File: appinfo/app.php (hunk 1 above)
Comment (VicDeo) [inline appinfo/app.php]: Not yet :) May be later

**Comment by VicDeo** [comment]
@nickvergessen Amendment is pushed. Thanks for reviewing

**Comment by nickvergessen** [comment]
fine by me

**Pull request merged**